src/Parser/FormUrlencoded.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace React\Http\Parser;
+
+use Evenement\EventEmitterTrait;
+use React\Http\Request;
+
+class FormUrlencoded implements ParserInterface
+{
+    use EventEmitterTrait;
+
+    /**
+     * @var string
+     */
+    protected $buffer = '';
+
+    /**
+     * @var Request
+     */
+    protected $request;
+
+    /**
+     * @var bool|integer
+     */
+    protected $contentLength = false;
+
+    /**
+     * @param Request $request
+     */
+    public function __construct(Request $request)
+    {
+        $this->request = $request;
+
+        $headers = $this->request->getHeaders();
+        $headers = array_change_key_case($headers, CASE_LOWER);
+
+        $this->contentLength = $headers['content-length'];
```

Comment:
Oh sorry, I meant that same line in RawBody :)

Regarding FormUrlencoded: Actually I think when the Content-Type is form-urlencoded but content-length is not set, then it means there is nothing to parse. Which is I guess perfectly valid. The http server should throw as less as possible exceptions as I think there are very less real constraints of such decoding.